crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -662,117 +693,168 @@ static const struct CipherTestVector {
-        false,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_192_ecb(),
         kAESKey_192,
         sizeof(kAESKey_192),
+        nullptr,
+        0,
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESECBCiphertext_192,
         sizeof(kAESECBCiphertext_192),
-        false,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_256_ecb(),
         kAESKey_256,
         sizeof(kAESKey_256),
+        nullptr,
+        0,
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESECBCiphertext_256,
         sizeof(kAESECBCiphertext_256),
-        false,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_128_cbc(),
         kAESKey,
         sizeof(kAESKey),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESCBCCiphertext,
         sizeof(kAESCBCCiphertext),
-        true,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_192_cbc(),
         kAESKey_192,
         sizeof(kAESKey_192),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESCBCCiphertext_192,
         sizeof(kAESCBCCiphertext_192),
-        true,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_256_cbc(),
         kAESKey_256,
         sizeof(kAESKey_256),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESCBCCiphertext_256,
         sizeof(kAESCBCCiphertext_256),
-        true,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_128_ctr(),
         kAESKey,
         sizeof(kAESKey),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESCTRCiphertext,
         sizeof(kAESCTRCiphertext),
-        true,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_192_ctr(),
         kAESKey_192,
         sizeof(kAESKey_192),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESCTRCiphertext_192,
         sizeof(kAESCTRCiphertext_192),
-        true,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_256_ctr(),
         kAESKey_256,
         sizeof(kAESKey_256),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESCTRCiphertext_256,
         sizeof(kAESCTRCiphertext_256),
-        true,
         AWSLC_APPROVED,
     },
     {
         EVP_aes_128_ofb(),
         kAESKey,
         sizeof(kAESKey),
+        kAESIV,
+        sizeof(kAESIV),
+        kPlaintext,
+        sizeof(kPlaintext),
         kAESOFBCiphertext,
         sizeof(kAESOFBCiphertext),
-        true,
         AWSLC_NOT_APPROVED,
     },
+    {
+        EVP_aes_256_xts(),
+        kAESXTSKey_256,
+        sizeof(kAESXTSKey_256),
+        kAESXTSIV_256,
+        sizeof(kAESXTSIV_256),
+        kAESXTSPlaintext_256,
+        sizeof(kAESXTSPlaintext_256),
+        kAESXTSCiphertext_256,
+        sizeof(kAESXTSCiphertext_256),
+        AWSLC_APPROVED,
+    },
```

Comment:
Can we add a test vector with AWSLC_NOT_APPROVED as the expected result? For example, this negative test case: https://github.com/Taffer/aws-lc/blob/4e1fd6c8f10d21a374a6539b8f4d93c26b0e88ad/crypto/fipsmodule/modes/xts_test.cc#L543